sos/plugins/__init__.py

```diff
@@ -677,6 +678,69 @@ def check_ext_prog(self, prog):
+                         binary=False, prepend_path=None, whitelist=[],
+                         blacklist=[]):
+        """Run a command or list of commands against storage-related devices.
+
+        Any commands specified by cmd will be iterated over the list of the
+        specified devices. Commands passed to this should include a '%(dev)s'
+        variable for substitution.
+
+            devices - Either 'block', 'fibre', or a list of device paths to
+                      run against. If set to 'block' or 'fibre', the commands
+                      will be run against the matching list of discovered
+                      devices.
+            prepend_path - the leading path for block device names, e.g.
+                        '/dev/' or '/sys/block/'.
+            whitelist - limit the devices iterated over. May be a str or list
+                        of strings (e.g. regexes)
+            blacklist - exclude the devices iterated over. May be a tr or list
+                        of strings (e.g. regexes)
+        """
+        if devices == 'block':
+            prepend_path = prepend_path or '/dev/'
+            devices = self.devices['block']
+        if devices == 'fibre':
+            devices = self.devices['fibre']
+        self._add_device_cmd(cmds, devices, timeout=timeout,
+                             sizelimit=sizelimit, chroot=chroot, runat=runat,
+                             env=env, binary=binary, prepend_path=prepend_path,
+                             whitelist=whitelist, blacklist=blacklist)
+
+    def _add_device_cmd(self, cmds, devices, timeout=300, sizelimit=None,
+                        chroot=True, runat=None, env=None, binary=False,
+                        prepend_path=None, whitelist=[], blacklist=[]):
+        """Run a command against all specified devices on the system.
+        """
+        if isinstance(cmds, six.string_types):
+            cmds = [cmds]
+        if isinstance(devices, six.string_types):
+            devices = [devices]
+        if whitelist and isinstance(whitelist, six.string_types):
+            whitelist = [whitelist]
+        if blacklist and isinstance(blacklist, six.string_types):
+            blacklist = [blacklist]
+        for cmd in cmds:
+            for device in devices:
+                _dev_ok = True
+                if whitelist:
+                    if not any(re.match(wl, device) for wl in whitelist):
+                        _dev_ok = False
+                if blacklist:
+                    if any(re.match(blist, device) for blist in blacklist):
+                        _dev_ok = False
+                if not _dev_ok:
+                    continue
+                if prepend_path:
+                    device = os.path.join(prepend_path, device)
+                sizelimit = sizelimit or self.get_option('log_size')
+                _cmd = cmd % {'dev': device}
+                self._add_cmd_output(_cmd, timeout=timeout,
```

Comment:
Maybe calling _log_debug similarly like deleted "`collecting udev info for:`" one? (though this method is more general and another debug will be printed in `_add_cmd_output` either way - the debug would have sense only if one is interested to know where the call comes from)